src/app/auth.context.tsx

```diff
@@ -137,6 +138,25 @@ function AuthProviderWrapper({ children, allMonths }: { children: React.ReactNod
     }
   };
 
+  const updateMonthIncomeExpenseDeletion = async (incomeExpenseId: string, monthId: string, isExpense: boolean) => {
+    const monthToUpdate = userMonths.find((oneMonth) => oneMonth._id === monthId);
+    const filteredMonths = userMonths.filter((oneMonth) => oneMonth._id !== monthId);
+    if (monthToUpdate) {
+      if (isExpense) {
+        const filteredMonthExpenses = monthToUpdate.expenses.filter((oneExpense) => oneExpense._id !== incomeExpenseId);
+        monthToUpdate.expenses = filteredMonthExpenses;
+      } else {
+        const filteredMonthIncomes = monthToUpdate.incomes.filter((oneIncome) => oneIncome._id !== incomeExpenseId);
+        monthToUpdate.incomes = filteredMonthIncomes;
+      }
+      // updates userMonths and UserYears
+      const updatedUserMonths = [...filteredMonths, monthToUpdate];
+      setUserMonths(updatedUserMonths);
+      const yearsData = await getYearsData(updatedUserMonths);
+      setUserYears(yearsData);
+    }
+  };
+
```

Comment:
I would move this to another context because it is not related to `auth context`